### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by dgageot**

### CODE REVIEW

**Comment by dgageot** [comment]
@briandealwis Could you test this PR with your sample that loses logs?

File: pkg/skaffold/kubernetes/log.go [hunk 1]
```diff
@@ -156,16 +157,26 @@ func (a *LogAggregator) streamContainerLogs(ctx context.Context, pod *v1.Pod, co
...
 	}()
+
+	headerColor := a.colorPicker.Pick(pod)
+	prefix := prefix(pod, container)
+	if err := a.streamRequest(ctx, headerColor, prefix, tr); err != nil {
+		logrus.Errorf("streaming request %s", err)
+	}
+
+	finalState := <-a.finalState[container.ContainerID]
+	a.printLogLine(headerColor, prefix, fmt.Sprintf("%s, exit code: %d %s\n", finalState.Reason, finalState.ExitCode, finalState.Message))
```
Comment (briandealwis) [inline pkg/skaffold/kubernetes/log.go]: I initially thought this "Completed, exit code: 0" message was from my program

File: pkg/skaffold/runner/build.go [hunk 2]
```diff
@@ -29,6 +29,22 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+func (r *SkaffoldRunner) buildAndTest(ctx context.Context, out io.Writer, artifacts []*latest.Artifact) error {
```
Comment (briandealwis) [inline pkg/skaffold/runner/build.go]: A bit confusing to have `BuildAndTest` and `buildAndTest`?

Comment (dgageot) [reply pkg/skaffold/runner/build.go]: Yes, I know. I tried hard to simplify but need to try harder

**Comment by dgageot** [comment]
Let's close this one and come back with another one next week after my other PRs are merged

**Pull request closed without merging**